src/index.ts

```diff
@@ -176,3 +189,122 @@ export const createInstance = (config: Config): Instance => {
+        config.contractAddress
+      );
+      const price = await contract.salePrice();
+      return ethers.utils.formatEther(price).toString();
+    },
+    getSaleData: async (): Promise<ClaimWithChildSaleData> => {
+      const contract = await getClaimContract(
+        config.web3Provider,
+        config.contractAddress
+      );
+
+      // always eth sales currently
+      const saleData: ClaimWithChildSaleData =
+        await claimWithChildSaleActions.getSaleData(contract, true);
+      return saleData;
+    },
+    getSaleStartBlock: async (): Promise<string> => {
+      const contract = await getClaimContract(
+        config.web3Provider,
+        config.contractAddress
+      );
+      const startBlock = await contract.saleStartBlock();
+      return startBlock.toString();
+    },
+    getSaleStatus: async (): Promise<SaleStatus> => {
+      const contract = await getClaimContract(
+        config.web3Provider,
+        config.contractAddress
+      );
+      const status: SaleStatus = await claimWithChildSaleActions.getSaleStatus(
+        contract
+      );
+      return status;
+    },
+    getSaleDuration: async (): Promise<ethers.BigNumber> => {
+      const contract = await getClaimContract(
+        config.web3Provider,
+        config.contractAddress
+      );
+      const duration = await contract.saleDuration();
+      return duration;
+    },
+    getTotalForSale: async (): Promise<ethers.BigNumber> => {
+      const contract = await getClaimContract(
+        config.web3Provider,
+        config.contractAddress
+      );
+      const total = await contract.totalForSale();
+
+      return total;
+    },
+    getNumberOfDomainsSold: async (): Promise<ethers.BigNumber> => {
+      const contract = await getClaimContract(
+        config.web3Provider,
+        config.contractAddress
+      );
+      const domainsSold = await contract.domainsSold();
+      return domainsSold;
+    },
+    getBlockNumber: async (): Promise<number> => {
+      const blockNum = await config.web3Provider.getBlockNumber();
+      return blockNum;
+    },
+    getEthBalance: async (address: string): Promise<string> => {
+      const balance = await config.web3Provider.getBalance(address);
+      return ethers.utils.formatEther(balance);
+    },
+    canBeClaimed: async (domainId: string): Promise<boolean> => {
+      const contract = await getClaimContract(
+        config.web3Provider,
+        config.contractAddress
+      );
+      const claimingAddress = await contract.domainsClaimedWithBy(domainId);
+      return claimingAddress == ethers.constants.AddressZero;
+    },
+    domainClaimedBy: async (domainId: string): Promise<string> => {
+      const contract = await getClaimContract(
+        config.web3Provider,
+        config.contractAddress
+      );
+      const claimingAddress = await contract.domainsClaimedWithBy(domainId);
+      return claimingAddress;
+    },
+    claimDomains: async (
+      claimingIds: string[],
```

Comment:
How does the SDK user get a list of claimingIds?
Can we expose a method that will return all of the ids that a user can claim with?

Reply:
Addressing this here - https://github.com/zer0-os/zSale-SDK/pull/27